### PR TITLE
Feature/merge 3.6

### PR DESCRIPTION
**Pull request by bdmogal**

### CODE REVIEW

**Review by wolf31o2 — APPROVED** [review]
👍 LGTM

**Comment by bdmogal** [comment]
Thanks @wolf31o2 . 

@sagarkapare could you please review `MapReduceContainerLauncher.java`. The rest of the conflicts were straight-forward.

**Comment by sagarkapare** [comment]
LGTM

**Comment by bdmogal** [comment]
Waiting for http://builds.cask.co/browse/CDAP-DUT4840-4

File: cdap-docs/vars [hunk 1]
```diff
@@ -49,6 +49,9 @@ GIT_BRANCH_CDAP_INGEST="1.3.0"
 GIT_BRANCH_CDAP_APPS="develop"
 GIT_VERSION_CDAP_APPS="0.8.0"
 
+# Used by examples manual
+GIT_BRANCH_CDAP_GUIDES="release/cdap-3.5-compatible" # Using until 3.6 versions available
+
```
Comment (awholegunch) [inline cdap-docs/vars:54]: This should be:

```
GIT_BRANCH_CDAP_GUIDES="develop"
```

and then PR https://github.com/caskdata/cdap/pull/6839 can be closed.

Comment (bdmogal) [reply cdap-docs/vars:54]: @awholegunch please see latest commit

**Comment by bdmogal** [comment]
Build failed due to flaky tests. Since this is just a merge PR, not going to run another build. Merging.

**Pull request merged**